### PR TITLE
[#4] Set up CD for iOS (TestFlight)

### PR DESCRIPTION
**Pull request by Shayokh144**
#4 
## What happened 👀
- Remove unused scripts
- Add separate scripts `ios_deploy_production.yml` and `ios_deploy_staging.yml` for production and staging app
- Add github secrets necessary for connecting appstoreconnect
- Update fastlane version to resolve transporter issue
- Update CODE_SIGN_IDENTITY
- Update iOS version from 10.0 to 11.0 to fix version mismatch error while creating ipa for testflight
- Update `Fastfile`, `Constant` file, `Gym` file
- Test `testflight` builds in physical iPhone device

## Insight 📝

- Update `fastlane scripts` and `xcode signing options` to upload production and staging app to testflight and distribute automatically to registered testers

## Proof Of Work 📹

- Production app
prod

- Staging app
staging

### CODE REVIEW

**Comment by luongvo** [comment]
@Shayokh144 @Tuubz Please make use of Draft PR for WIP work 🙏

File: .github/workflows/ios_deploy_to_testflight_production.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: ios-deploy-to-testflight-production
+on:
+  # Trigger the workflow on push action
+  workflow_dispatch:
```
Comment (Tuubz) [inline .github/workflows/ios_deploy_to_testflight_production.yml:4]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

> To enable a workflow to be triggered manually

If I'm not mistaken, this is redundant 🙏

File: ios/fastlane/Constants/Constants.rb [hunk 2]
```diff
@@ -67,21 +67,21 @@ def self.SCHEME_NAME_PRODUCTION
 
   # an staging product name
   def self.PRODUCT_NAME_STAGING
-    'Flutter Template Staging'
+    'Runner'
```
Comment (Tuubz) [inline ios/fastlane/Constants/Constants.rb:70]: Any reason it's called `Runner`? 🙇

File: .github/workflows/ios_deploy_to_testflight_production.yml
Comment (Tuubz) [inline .github/workflows/ios_deploy_to_testflight_production.yml:1]: Optional renaming:

- `ios_deploy_to_staging.yml`
- `ios_deploy_to_production.yml`

File: .github/workflows/ios_deploy_to_testflight_staging.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      uses: subosito/flutter-action@v2
+      with:
+        channel: 'stable'
+        flutter-version: '3.3.10'
+
+    - name: Create Env file
+      run: |
+        touch .env.staging
+        echo "SECRET=\"MY_SECRET\"" >> .env.staging
+        echo "REST_API_ENDPOINT=\"MY_ENDPOINT\"" >> .env.staging
```
Comment (Tuubz) [inline .github/workflows/ios_deploy_to_testflight_staging.yml:39]: I think it should be updated to this, right? (Please refer to `android_deploy_staging.yml`) 🙏 

```suggestion
      - name: Setup .env.staging
        env:
          ENV_STAGING: ${{ secrets.ENV_STAGING }}
        run: |
          echo "$ENV_STAGING" > .env.staging
```

File: .github/workflows/ios_deploy_to_testflight_staging.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    - name: Deploy to TestFlight
+      run: cd ./ios && bundle exec fastlane futter_build_and_upload_testflight_staging_app
+      env:
+        TEAM_ID: ${{ secrets.IOS_TEAM_ID }}
+        APP_STORE_CONNECT_API_KEY_KEY_ID: ${{ secrets.APP_STORE_CONNECT_API_KEY_KEY_ID }}
+        APP_STORE_CONNECT_API_KEY_IS_KEY_CONTENT_BASE64: ${{ secrets.APP_STORE_CONNECT_API_KEY_IS_KEY_CONTENT_BASE64 }}
+        APP_STORE_CONNECT_API_KEY_ISSUER_ID: ${{ secrets.APP_STORE_CONNECT_API_KEY_ISSUER_ID }}
+        SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+        MATCH_REPO: ${{ secrets.MATCH_REPO }}
+        MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
```
Comment (Tuubz) [inline .github/workflows/ios_deploy_to_testflight_staging.yml:71]: This looks redundant with line 14-20 💭 

Is it possible this is redundant?

File: ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -367,21 +366,26 @@
...
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
-				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
-				DEVELOPMENT_TEAM = 4TWS7E2EPE;
+				CODE_SIGN_IDENTITY = "iPhone Distribution";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Distribution";
+				CODE_SIGN_STYLE = Manual;
+				CURRENT_PROJECT_VERSION = 21;
+				DEVELOPMENT_TEAM = "";
+				"DEVELOPMENT_TEAM[sdk=iphoneos*]" = 4TWS7E2EPE;
```
Comment (Tuubz) [inline ios/Runner.xcodeproj/project.pbxproj:377]: Just curious, are these necessary? 

If so, they should be added to the templates by default

cc @luongvo

**Review by Tuubz — COMMENTED** [review]
Damn, a lot more changes then I expected 😱 💪

File: ios/Runner.xcodeproj/project.pbxproj (hunk 5 above)
Comment (Shayokh144) [reply ios/Runner.xcodeproj/project.pbxproj:377]: @Tuubz first I tried with CODE_SIGN_STYLE = Automatic; but Fastlane throw provision profile related error for that.

File: ios/fastlane/Constants/Constants.rb (hunk 2 above)
Comment (Shayokh144) [reply ios/fastlane/Constants/Constants.rb:70]: @Tuubz This is the default name of the project but if you install the app from testflight you will see name `SurveyProduction` or `SurveyStaging`

**Comment by Shayokh144** [comment]
@Tuubz I updated according to your comment, please review again when you are free.

**Review by Tuubz — APPROVED** [review]
LGTM 🍎

File: ios/Runner/Info.plist [hunk 6]
```diff
@@ -22,7 +21,7 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>$(FLUTTER_BUILD_NUMBER)</string>
+	<string>21</string>
```
[the remark this answers is not in the document]
Comment (Shayokh144) [reply ios/Runner/Info.plist:24]: This was changed automatically when I run test on uploading to testflight from my laptop, will update this soon.

**Review by luongvo — APPROVED** [review]

**Pull request merged**